lib/commands/misc/giverole.js

```diff
@@ -24,46 +24,80 @@ module.exports = class GiveRoleCommand extends Command {
 		this.queues = {};
 	};
 
+	grantRole(msg, role, channel) {
+		msg.member.addRole(role);
+		channel.send(`${msg.author} now has the ${role.name} role!`);
+	}
+
+	giveMemberRole(msg, role, channel, userRoles) {
+		// Only allows this to run if the user is not already a Member
+		if (!userRoles.has(role.id)) {
+			msg.member.addRole(role);
+
+			channel.send(`Welcome to the server, ${msg.author}! <:wutGiga:297897855727697921>`);
+		}
+	}
+
+	giveColorRole(msg, role, channel, userRoles, memberRole) {
+		// Confirms that the user doesn't already have the color role and is a Member
+		if (!userRoles.has(role.id) && userRoles.has(memberRole.id)) {
+
+			// Prevents the user from claiming more than one color role
+			if (!userRoles.some(r => COLOR_ROLES.includes(r.name))) {
+				this.grantRole(msg, role, channel);
+			} else {
+				msg.reply("You have already claimed another color role! Use r!rmrole [role name] and try again.");
+			}
+
+		} else {
+		  msg.reply("You have already claimed that role! Use r!rmrole [role name] and try again.");
+		}
+	}
+
+	giveRegionRole(msg, role, channel, userRoles, memberRole) {
+		// Confirms that the user doesn't already have the region role and is a Member
+		if (!userRoles.has(role.id) && userRoles.has(memberRole.id)) {
+
+			// Prevents the user from claiming more than one region role
+			if (!userRoles.some(r => REGION_ROLES.includes(r.name))) {
+				this.grantRole(msg, role, channel);
+			} else {
+				msg.reply("You have already claimed another regional role! Use r!rmrole [role name] and try again.");
+			}
+
+		} else {
+		  msg.reply("You have already claimed that role! Use r!rmrole [role name] and try again.");
+		}
+	}
+
 	async run(msg, args) {
 		const roleArg = args.role;
-		const memberRoles = msg.member.roles;
+		const userRoles = msg.member.roles;
+		const memberRole = msg.guild.roles.find(r => r.name === 'Member');
+		const generalChannel = this.client.channels.get(getChannel(GENERAL_ID));
+		const botChannel = this.client.channels.get(getChannel(BOT_STUFF_ID));
 
     if (roleArg.toLowerCase() === "member") {
       const role = msg.guild.roles.find(r => r.name === 'Member');
+			this.giveMemberRole(msg, role, generalChannel, userRoles);
 
-      if (!memberRoles.has(role.id)) {		// Needs to use role.id to properly check if the user has that role -Syn
-        msg.member.addRole(role);
-        const channel = this.client.channels.get(getChannel(GENERAL_ID));
-        channel.send(`Welcome to the server, ${msg.author}! <:wutGiga:297897855727697921>`);
-      }
+    } else if (COLOR_ROLES.includes(roleArg) || REGION_ROLES.includes(roleArg)) {
```

Comment:
The if statements here can be cleaner:

maybe something like:
1. if the argument is member, `giveMemberRole`
2. if not, return if it is not in the command channel.

1. If it is a COLOR_ROLE, `giveColorRole`
2. else if it is a REGION_ROLE, `giveRegionRole`
3. otherwise, it is not a valid role.